src/NuGet.Clients/VsConsole/PowerShellHost/PowerShellHost.cs

```diff
@@ -399,7 +399,7 @@ private void UpdateWorkingDirectory()
                 // projects might not be loaded when DPL is enabled.
                 _solutionManager.EnsureSolutionIsLoaded();
 
-                if (!_solutionManager.IsSolutionFullyLoaded)
+                if (!await _solutionManager.IsSolutionFullyLoaded())
```

Comment:
Here, `PowerShellHost` will switch to the main thread. This may cause a deadlock.